src/components/Calculator.js

```diff
@@ -11,67 +11,219 @@ const CalculatorLayout = styled.div`
-    <Button backgroundColor="orange" color="white">
-      -
-    </Button>
-    <Button backgroundColor="gainsboro" color="black">
-      1
-    </Button>
-    <Button backgroundColor="gainsboro" color="black">
-      2
-    </Button>
-    <Button backgroundColor="gainsboro" color="black">
-      3
-    </Button>
-    <Button backgroundColor="orange" color="white">
-      +
-    </Button>
-    <Button backgroundColor="gainsboro" size="large" color="black">
-      0
-    </Button>
-    <Button backgroundColor="gainsboro" color="black">
-      ,
-    </Button>
-    <Button backgroundColor="orange" color="white">
-      =
-    </Button>
-  </CalculatorLayout>
-);
+const Calculator = () => {
+  const [result, setResult] = useState("0");
+  const [operation, setOperation] = useState("noOp");
+  const [firstOperand, setFirstOperand] = useState(0);
+  const [secondOperand, setSecondOperand] = useState(0);
+  const [isReset, setResetStatus] = useState(true);
+
+  const onResetClick = value => {
+    if (result !== "0") setResult("0");
+    if (firstOperand !== 0) setFirstOperand(0);
+    if (secondOperand !== 0) setSecondOperand(0);
+    if (operation !== "noOp") setOperation("noOp");
+    if (isReset === false) setResetStatus(true);
+  };
+
+  const onNumberClick = value => {
+    setResetStatus(false);
+    if (value === ",") setResult(!result.includes(".") ? result + "." : result);
+    else {
+      if (firstOperand !== 0 || result === "0.") {
+        setResult(result + value);
+        setFirstOperand(Number(result + value));
+      } else {
+        setResult(value);
+        setFirstOperand(Number(value));
+      }
+    }
+  };
+
+  const onUnaryOperatorClick = value => {
+    switch (value) {
+      case "%":
+        setSecondOperand(Number(result) / 100);
+        setResult((Number(result) / 100).toString());
+        break;
+      case "±":
+        setSecondOperand(-Number(result));
+        setResult((-Number(result)).toString());
+        break;
+      default:
+    }
+    setFirstOperand(0);
+    setOperation(value);
+  };
+
+  const onArithOperatorClick = value => {
+    switch (operation) {
+      case "+":
+        setSecondOperand(secondOperand + firstOperand);
+        setResult(secondOperand + firstOperand);
+        break;
+      case "-":
+        setSecondOperand(secondOperand - firstOperand);
+        setResult(secondOperand - firstOperand);
+        break;
+      case "×":
+        setSecondOperand(secondOperand * firstOperand);
+        setResult(secondOperand * firstOperand);
+        break;
+      case "÷":
+        setSecondOperand(secondOperand / firstOperand);
+        setResult(secondOperand / firstOperand);
+        break;
+      case "noOp":
+        setSecondOperand(firstOperand);
+        break;
+      default:
+    }
+    setFirstOperand(0);
+    setOperation(value);
+  };
+
+  const reset = isReset === true ? "AC" : "C";
+
+  return (
+    <CalculatorLayout>
+      <Display result={result} />
+      <Button
```

Comment:
You can create a config array for the buttons with the props each needs and then iterate through it and render all of them at once.